src/xenia/kernel/xam/content_manager.cc

```diff
@@ -247,7 +290,11 @@ X_RESULT ContentManager::DeleteContent(const XCONTENT_DATA& data) {
 
   auto package_path = ResolvePackagePath(data);
   if (xe::filesystem::PathExists(package_path)) {
-    xe::filesystem::DeleteFolder(package_path);
+    if (xe::filesystem::IsFolder(package_path)) {
```

Comment:
Okay. Conditional logic like this is a bad idea, perhaps we need to make a base class and derive into multiple classes for a folder/package/etc?

Reply:
Yeah that would be a lot better, I'll have a try seeing if I can get something like that to work now.